avoid a warning:

  CC      drivers/acpi/acpica/evmisc.o
  CC      lib/iov_iter.o
In file included from include/linux/linkage.h:4:0,
                 from include/linux/kernel.h:6,
                 from include/linux/uio.h:12,
                 from lib/iov_iter.c:2:
lib/iov_iter.c: In function 'import_single_range':
./arch/x86/include/asm/uaccess.h:112:14: warning: logical not is only applied to the left hand side of comparison [-Wlogical-not-parentheses]
     if (type != VERIFY_WRITE) {  \
              ^
include/linux/compiler.h:186:42: note: in definition of macro 'unlikely'
 # define unlikely(x) __builtin_expect(!!(x), 0)
                                          ^
lib/iov_iter.c:844:16: note: in expansion of macro 'access_ok'
  if (unlikely(!access_ok(!rw, buf, len)))
                ^
  CC      net/core/stream.o


--- orig/arch/x86/include/asm/uaccess.h	2015-09-14 17:31:10.486794568 +0200
+++ patched/arch/x86/include/asm/uaccess.h	2015-09-14 17:31:34.100493081 +0200
@@ -109,7 +109,7 @@ extern int _cond_resched(void);
 					_cond_resched();		\
 				if (__get_user(__c_ao, (char __user *)__addr))	\
 					break;				\
-				if (type != VERIFY_WRITE) {		\
+				if ((type) != VERIFY_WRITE) {		\
 					__addr = __addr_ao;		\
 					continue;			\
 				}					\
